opm/autodiff/RedistributeDataHandles.hpp

```diff
@@ -288,4 +306,54 @@ class BlackoilPropsDataHandle
     std::size_t size_;
 };
 
+inline
+void distributeGridAndData( Dune::CpGrid& grid,
```

Comment:
Is there a reason why you used an overload instead of a template specialisation here? I think that using overloads it is better not to have the general template (i.e. having `UnstructuredGrid` as parameter on line 36).